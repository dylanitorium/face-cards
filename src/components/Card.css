.card {
  cursor: pointer;
  box-sizing: border-box;
  border: 2px solid #EEEEEE;
  padding: 50px 20px;
  color: #294762;
  font-family: 'Open Sans', sans-serif;
  font-size: 13px;
  line-height: 18px;
  box-shadow: 2px 2px 1px 0px rgba(238,238,238,0.6);
  border-radius: 2px;
}

.card::before,
.card::after {
  content: '';
  display: table;
}

.card::after {
  clear: both;
}

.active {
  border-color: #D8D8D8;
}

.avatar {
  float: left;
  width: 104px;
  height: 104px;
  border-radius: 2px;
}

.content {
  box-sizing: border-box;
  float: left;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 0 0 0 20px;
  width: 186px;
  height: 104px;
}

.quote {
  font-style: italic;
}

.details {
  margin-bottom: 5px;
}

.name {
  font-weight: 600;
}
