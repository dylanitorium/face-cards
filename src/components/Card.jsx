import React from 'react';
import PropTypes from 'prop-types';
import card from './Card.css';

const Card = props =>  (
  <div onClick={() => props.onCardClick(props.user.id)} className={props.user.id === props.currentUserId ? `${card.card} ${card.active}`  : card.card}>
    <img className={card.avatar} src={`https://api.adorable.io/avatars/150/${props.user.email}.png`}/>
    <div className={card.content}>
      <div className={card.quote}>
        "{props.user.company.catchPhrase}"
      </div>
      <div className={card.details}>
        <div className={card.name}>
          {props.user.name}
        </div>
        <div className={card.email}>
          {props.user.email}
        </div>
      </div>
    </div>
  </div>
);

Card.propTypes = {};

export default Card;
