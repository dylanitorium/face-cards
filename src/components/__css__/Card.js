import styled from 'styled-components';

const StyledCard = styled.div`
  border: 1px solid black;
`;

export default StyledCard;