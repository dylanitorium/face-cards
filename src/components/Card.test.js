import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';
import { user } from '../fixtures';

test('Renders correctly', () => {
  const component = renderer.create(
    <Card user={user} onCardClick={() => {}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders correctly when is current user', () => {
  const component = renderer.create(
    <Card user={user} currentUserId={user.id} onCardClick={() => {}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
