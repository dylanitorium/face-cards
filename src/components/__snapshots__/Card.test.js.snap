// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className={undefined}
  onClick={[Function]}
>
  <img
    className={undefined}
    src="https://api.adorable.io/avatars/150/Sincere@april.biz.png"
  />
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      "
      Multi-layered client-server neural-net
      "
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        Leanne Graham
      </div>
      <div
        className={undefined}
      >
        Sincere@april.biz
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly when is current user 1`] = `
<div
  className="undefined undefined"
  onClick={[Function]}
>
  <img
    className={undefined}
    src="https://api.adorable.io/avatars/150/Sincere@april.biz.png"
  />
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      "
      Multi-layered client-server neural-net
      "
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        Leanne Graham
      </div>
      <div
        className={undefined}
      >
        Sincere@april.biz
      </div>
    </div>
  </div>
</div>
`;
